### PR TITLE
Implement protobuf codec for case class schemas

### PR DESCRIPTION
**Pull request by thinkharderdev**

### CODE REVIEW

File: core/src/test/scala/zio/schema/codec/JsonCodecSpec.scala [hunk 1]
```diff
@@ -109,7 +109,7 @@ object JsonCodecSpec extends DefaultRunnableSpec {
         assertEncodes(
           adtSchema,
           Enumeration(StringValue("foo")),
-          JsonCodec.Encoder.charSequenceToByteChunk("""{"value":{"string":"foo"}}""")
+          JsonCodec.Encoder.charSequenceToByteChunk("""{"oneOf":{"StringValue":{"value":"foo"}}}""")
```
Comment (thinkharderdev) [inline core/src/test/scala/zio/schema/codec/JsonCodecSpec.scala:112]: @jdegoes I'm not 100% sure about this but I think the encoding before was wrong w/r/t case classes with ADT fields. Not sure if the way it was implemented was just an optimization though....

File: core/src/test/scala/zio/schema/codec/ProtobufCodecSpec.scala [hunk 2]
```diff
@@ -64,21 +64,21 @@ object ProtobufCodecSpec extends DefaultRunnableSpec {
...
+          e  <- encode(Record.schemaRecord, Record("Foo", 123)).map(toHex)
+          e2 <- encodeNS(Record.schemaRecord, Record("Foo", 123)).map(toHex)
         } yield assert(e)(equalTo("0A03466F6F107B")) && assert(e2)(equalTo("0A03466F6F107B"))
       },
       testM("enumerations") {
         for {
           e  <- encode(schemaEnumeration, Enumeration(IntValue(482))).map(toHex)
           e2 <- encodeNS(schemaEnumeration, Enumeration(IntValue(482))).map(toHex)
-        } yield assert(e)(equalTo("0A0310E203")) && assert(e2)(equalTo("0A0310E203"))
+        } yield assert(e)(equalTo("0A05120308E203")) && assert(e2)(equalTo("0A05120308E203"))
```
Comment (thinkharderdev) [inline core/src/test/scala/zio/schema/codec/ProtobufCodecSpec.scala:75]: Same thing here as above.

**Review by jdegoes — APPROVED** [review]

**Comment by jdegoes** [comment]
@thinkharderdev Looks good to merge after the conflict is fixed!

**Review by jdegoes — APPROVED** [review]

**Comment by jdegoes** [comment]
Thanks for your work on this! Protobuf continues to improve.

**Pull request merged**